src/server/game/Handlers/BattleGroundHandler.cpp

```diff
@@ -158,6 +158,9 @@ void WorldSession::HandleBattlemasterJoinOpcode(WorldPacket & recvData)
                  _player->InBattlegroundQueueForBattlegroundQueueType(BATTLEGROUND_QUEUE_3v3) ||
                  _player->InBattlegroundQueueForBattlegroundQueueType(BATTLEGROUND_QUEUE_5v5)) // can't be already queued for arenas
             err = ERR_BATTLEGROUND_QUEUED_FOR_RATED;
+        // don't let Death Knights join BG queues when they are not allowed to be teleported yet
+        else if (_player->getClass() == CLASS_DEATH_KNIGHT && _player->GetMapId() == 609 && !_player->IsGameMaster() && !_player->HasSpell(50977))
+            err = ERR_BATTLEGROUND_NONE;
```

Comment:
When Death Knight Queue solo, it will tell the player, unable to join the queue.
```suggestion
            err = ERR_BATTLEGROUND_JOIN_TIMED_OUT;
```

Reply:
It sends "Join as groupe failed"